custom_components/ha-glocaltokens/sensor.py

```diff
@@ -21,25 +29,60 @@ class GlocaltokensSensor(GlocaltokensEntity):
     def __init__(self, coordinator, entry, device):
         """Initialize the sensor."""
         super().__init__(coordinator, entry)
-        self._name = device[GLOCALTOKENS_DEVICE_NAME]
-        self._token = device[GLOCALTOKENS_TOKEN]
+        self._name = device[DEVICE_NAME]
+        self._state = device[TOKEN]
 
     @property
-    def name(self):
-        """Return the name of the sensor."""
-        return f"{DEFAULT_NAME} {self._name}"
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            "device": str(self.name),
+            "integration": DOMAIN,
+        }
 
-    @property
-    def state(self):
-        """Return the state of the sensor."""
-        return self._token
+class GlocaltokensAlarmSensor(GlocalAlarmEntity):
+    """Representation of a Sensor."""
+
+    def __init__(self, coordinator, entry, device):
+        """Initialize the sensor."""
+        super().__init__(coordinator, entry)
+        self._name = device[DEVICE_NAME]
+        self._alarms = device[ALARMS]
+
+        if len(self._alarms) > 0:
+            self._state = convert_from_ms_to_s(self._alarms[0][FIRE_TIME])
```

Comment:
I am not really sure about this one. So we are saying that if `device[ALARMS]` has more than one alarm set then we only take the first one from the list to represent in this sensor? How about the others? I believe this sensor is not really "Google Home alarm sensor" but more like "Google Home next alarm sensor", right?
We could either make it simply like that. But it would be great if we could support multiple alarms for the same google home device in the future, don't you think so?

Reply:
All the alarms are listed as attributes. 

This ways was made so that i could debug it, but my original intention with this was to make similar to how the next alarm sensor function with mobile app.

I think it should always show the next alarm when there is one and then list the rest as attributes. Or we could just create a new alarm sensor for each?

Reply:
I really like your idea about "the next alarm". But it would be hard to implement it that way i guess.
With google home, we can set two types of alarms: single and recurring. Selecting the first alarm from the list doesn't really mean it will be the next alarm. Imagine the list with two alarms where the first on is alarm for every day at 7am, while the second alarm is just at 5pm tonight. And if the now time is 1pm, the next alarm is not going to be the first element in the list, but the second. 

I am not sure what's the best way for us to ultimately go with that but what if we end up implementing generic "GlocaltokenAlarm" for each of the alarms for each of the devices? And inside of the alarm, we can have attributes like: `recurring` and `reccurring_days`. For recurring alarms they will be `recurring = True; recurring_days = "..."` and for single alarms `recuring = False; recurring_days = None`. And then on top of it all we can have a `service` that will be called something like `get_next_alarm`, with input argument of google home's name. Which will go through all Alarm sensors in that particular Google Home's alarms and find the next alarm by comparing them all? 

But even if so, let's just keep it as you have already implemented and we can take that discussion as a whole separate ticket afterwards :)

Reply:
First part. Your are absolut right, it was actually what i ment in my first reply, but i did't do a good job of explaining it.
We should sort through all alarms and the choose the one with closes fire_time to now().

The problem with a sensor for each alarm/sensor is that you dont know what the entity name is in HA when you try to call it in an automation or make a card (which i have done). I will create a separate issue for this, so we can find a good way to implement it. :smile:

Reply:
Oh nice work with the card! 💥  Do you feel like hosting it separately or under the same project?